src/Common/src/System/SR.vb

```diff
@@ -92,3 +103,23 @@ Namespace System
         End Function
     End Class
 End Namespace
+
+Namespace System.Resources.Diagnostics
+
+    <AttributeUsage(AttributeTargets.All)>
+    <ExcludeFromCodeCoverage>
+    Friend Class ExcludeFromCodeCoverageAttribute
+        Inherits Attribute
+
+        ' The code in the partial SR above is injected into all assemblies with resources, regardless of
+        ' whether those assemblies use this functionality.  As such, it should be excluded from code coverage.
+        ' The code coverage tools are configured to ignore any code attributed with an attribute
+        ' named ExcludeFromCodeCoverage, regardless of its namespace.  We have it in a specialized namespace
+        ' so as to avoid conflicts with other ExcludeFromCodeCoverageAttribute types used elsewhere in corefx.
+        ' It's applied to the individual members in SR rather than to SR as a whole because we still
+        ' want code coverage to include the individual resource keys; doing so helps to highlight whether
+        ' we're exercising all error paths, whether any resource strings are stale and can be removed, etc.
+
+    End Class
+
+End Namespace
```

Comment:
> do we need to exclude only these methods or the whole SR class. I am just wondering if we need to set the attribute on the whole class instead.

If we wanted to exclude the whole SR class, we could do so without code changes, by just modifying the command-line arguments passed to the code coverage tool to exclude the SR class.  But I don't think we should do that.  There's value in knowing which of the resource strings are being accessed, via seeing whether the associated generated properties are hit during testing: it helps to highlight stale strings that can be removed, and it helps to highlight error paths that are never being used.  Unfortunately all of these helpers and those resource strings are on the same class.

> Could we use an existing attribute like GeneratedCodeAttribute or EditorBrowsableAttribute or something of that nature.

GeneratedCodeAttribute is a good idea, but the compiler puts it more places than you might expect, e.g. the state machine type that's generated for async methods is annotated as GeneratedCode, which means that effectively all async methods would end up being excluded from code coverage.  EditorBrowsableAttribute could work, but just because something is advanced or meant to be internal and should be hidden from users in VS doesn't mean we don't want that code to be tested, e.g. https://github.com/dotnet/corefx/blob/41e203011152581a6c65bb81ac44ec037140c1bb/src/System.IO.Compression.ZipFile/src/System/IO/Compression/ZipFileExtensions.cs#L9-L10 .  I'm not aware unfortunately of any other attributes that could be used without inadvertently suppressing code that should actually be tested.

> it would just be nice if we didn't need to litter our code with yet more custom attributes.

I may have found another solution.  I won't merge this until I've investigated that.  We might be able to just filter out this specific file from coverage with a change to build tools.